lnwallet/channel.go

```diff
@@ -951,8 +951,10 @@ func (lc *LightningChannel) diskHtlcToPayDesc(feeRate chainfee.SatPerKWeight,
 		BlindingPoint:      htlc.BlindingPoint,
 	}
 
-	// Ensure that we'll restore any custom records which were stored as
-	// extra data on disk.
+	// When the HTLC was decoded from the wire, the ExtraData field was
+	// parsed and split into the BlindingPoint and CustomRecords. Which
+	// means we don't have to copy over the ExtraData itself into the
+	// PaymentDescriptor, just the encoded CustomRecords.
```

Comment:
I'm not sure if this comment is true. The `ExtraData` field still might contain data for a field which is lower in type than the custom records type range but not a defined field on the HTLC.

Reply:
True. But we currently ignore those left-over values and don't copy them into the payment descriptor (we don't have a field there). It IMO didn't make sense to encode/decode the custom records from/to the blob. So I merged this commit with https://github.com/lightningnetwork/lnd/pull/8771/commits/d718a5bdf76baeb332bb8e150ad994988406bd07, which also gets rid of this possibly incorrect comment.